components/WrapPanel2/src/WrapPanel2.Properties.cs

```diff
@@ -0,0 +1,142 @@
+        nameof(ItemSpacing),
+        typeof(double),
+        typeof(WrapPanel2),
+        new PropertyMetadata(default(double), OnPropertyChanged));
+
+    /// <summary>
+    /// Backing <see cref="DependencyProperty"/> for the <see cref="LineSpacing"/> property.
+    /// </summary>
+    public static readonly DependencyProperty lineSpacingProperty = DependencyProperty.Register(
+        nameof(LineSpacing),
+        typeof(double),
+        typeof(WrapPanel2),
+        new PropertyMetadata(default(double), OnPropertyChanged));
+
+    /// <summary>
+    /// Backing <see cref="DependencyProperty"/> for the <see cref="FixedRowLengths"/> property.
+    /// </summary>
+    public static readonly DependencyProperty FixedRowLengthsProperty = DependencyProperty.Register(
+        nameof(FixedRowLengths),
+        typeof(bool),
+        typeof(WrapPanel2),
+        new PropertyMetadata(default(bool), OnPropertyChanged));
+
+    /// <summary>
+    /// Backing <see cref="DependencyProperty"/> for the <see cref="StretchChildren"/> property.
+    /// </summary>
+    public static readonly DependencyProperty StretchChildrenProperty = DependencyProperty.Register(
+        nameof(StretchChildren),
+        typeof(StretchChildren),
+        typeof(WrapPanel2),
+        new PropertyMetadata(default(StretchChildren), OnPropertyChanged));
+
+    /// <summary>
+    /// Backing <see cref="DependencyProperty"/> for the <see cref="OverflowBehavior"/> property.
+    /// </summary>
+    public static readonly DependencyProperty OverflowBehaviorProperty = DependencyProperty.Register(
+        nameof(OverflowBehavior),
+        typeof(OverflowBehavior),
+        typeof(WrapPanel2),
+        new PropertyMetadata(default(OverflowBehavior), OnPropertyChanged));
+
+    /// <summary>
+    /// Gets or sets the panel orientation.
+    /// </summary>
+    public Orientation Orientation
+    {
+        get => (Orientation)GetValue(OrientationProperty);
+        set => SetValue(OrientationProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the spacing between items.
+    /// </summary>
+    public double ItemSpacing
+    {
+        get => (double)GetValue(ItemSpacingProperty);
+        set => SetValue(ItemSpacingProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the vertical spacing between items.
+    /// </summary>
+    public double LineSpacing
+    {
+        get => (double)GetValue(lineSpacingProperty);
+        set => SetValue(lineSpacingProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets whether or not all rows/columns should stretch to match the length of the longest.
+    /// </summary>
+    public bool FixedRowLengths
+    {
+        get => (bool)GetValue(FixedRowLengthsProperty);
+        set => SetValue(FixedRowLengthsProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the method used to fill rows without a star-sized item.
+    /// </summary>
+    public StretchChildren StretchChildren
```

Comment:
See discussion here: https://github.com/microsoft/microsoft-ui-xaml/pull/10858#discussion_r2615447331

We're updating to `ItemsStretch` to match UniformGridLayout.